### ISSUE TITLE
Modification of plot2D to allow filled characters

### ISSUE THREAD

**Issue by lmsimp** [lgatto/pRoloc#152]
I would like to add functionality to `plot2D` to allow the use of filled point characters. Currently, it is not easy to straightforward to implement with `...` which is passed to `points` in the `plot2D` code. 

I would explicitly like to add an argument `bg` to `plot2D` (as it is used in `points`) and add functions `getStockbg`, `setStockbg` to work in the same way we currently have for `col` and setting marker colours. 

I have already an amended version of `plot2D` I use that allows this and would like to submit a PR to update the version we have in `pRoloc`. 

As discussed with you at Bioc eventually we may default to using `ggplot` for generating maps but `plot2D` is still very much used day-to-day in the lab and I think it is a great function that works out of the box within the `pRoloc` workflow.

**Comment by lgatto** [comment]
Does your update break any existing code? If not, then, I think it could be added. If there more risky changes, unit tests that tests the previous and new behaviour would be good.

**Comment by lmsimp** [comment]
No existing code is broken. I think all is okay. I will check the vignettes.

I want to also change some of the default behaviour such as the call to `grid` and change it from `grid = TRUE` to `grid = FALSE`. Can I just change this or do I need some interim step to warn users the default behaviour has changed?

In any case, I will wait for @Charl-Hutchings to submit her PR for markers next week before adding this to a branch and submitting a PR.

**Comment by lgatto** [comment]
Not that (some of) the plot2D will hopefully become obsolete after the refactoring to SE, as it'll be possible to easily get a long table to use ggplot() on the fly.

**Comment by lmsimp** [comment]
Yes, it will be really great to be able to pipe output to ggplot on the fly. I have been thinking however it would be nice to keep plot2D and alongside using ggplot... but I do realise you are keen to lighten the code load and of avoid redundancy.
